### PR TITLE
Refactoring TPC-DS benchmark for SnappyData, Spark

### PR DESCRIPTION
**Pull request by cruizen**
## Changes proposed in this pull request

Refactored the tpc-ds benchmark scripts, classes for small improvements
* corrected the query names used in the conf file to point to query files
* spark caching made optional in spark perf test with a config param
* Using df.count instead of df.collect().length in table creation to remedy errors, slow execution
* Fixed some style warning, log messages, comments, dead code
* Fixed query names in tpc-ds template
* refactored tpc-ds spark conf file template and job, spark job
* fixed conf file templates for TPC-DS
* refactored tpc-ds query execution job to write structured output to csv files
* Changed customer_demographics and item tables to column tables

## Patch testing

By running the benchmark locally at a small scale (1g, 10g)

## ReleaseNotes.txt changes

None needed until/ unless we add docs for the user. 

## Other PRs 

(Does this change require changes in other projects- store, spark, spark-jobserver, aqp? Add the links of PR of the other subprojects that are related to this change)
No

### CODE REVIEW

**Comment by spillai14** [comment]
Looks good.

File: tests/benchmark/snappy/tpcds/spark/PerfRun.conf.template [hunk 1]
```diff
@@ -1,10 +1,7 @@
 #Do not change the way of queries, sparkProperties, sparkSqlProperties. Just change the values inside strings
 
 #queries to be run
-queries="1-2-3-4-5-6-7-8-9-10-11-12-13-14-15-16-17-18-19-20-22-21"
-#queries=10
-#queries="\"1-3-14\""
-
+queries="q1,q2,q3,q4,q5,q6,q7,q8,q9,q10,q11,q12,q13,q14a,q14b,q15,q16,q17,q18,q19,q20,q21,q22,q23a,q23b,q24a,q24b,q25,q26,q27,q28,q29,q30,q31,q32,q33,q34,q35,q36,q37,q38,q39a,q39b,q40,q41,q42,q43,q44,q45,q46,q47,q48,q49,q50,q51,q52,q53,q54,q55,q56,q57,q58,q59,q60,q61,q62,q63,q64,q65,q66,q67,q68,q69,q70,q71,q72,q73,q74,q75,q76,q77,q78,q79,q80,q81,q82,q83,q84,q85,q86,q87,q88,q89,q90,q91,q92,q93,q94,q95,q96,q97,q98,q99"
```
Comment (nikhilbandi) [inline tests/benchmark/snappy/tpcds/spark/PerfRun.conf.template]: Query names in template need to be corrected.

File: tests/benchmark/snappy/tpcds/spark/2_Setup_Table_Query.sh [hunk 2]
```diff
@@ -10,13 +10,10 @@ for element in "${slaves[@]}";
 echo "******************Created conf/slaves******************"
 
 #Start master and slaves from master machines
-ssh $master sh $SnappyData/sbin/start-all.sh
+ssh $master bash $SnappyData/sbin/start-all.sh
```
Comment (nikhilbandi) [inline tests/benchmark/snappy/tpcds/spark/2_Setup_Table_Query.sh]: Please check if the script can be run directly without specifying shell.

File: cluster/src/test/scala/io/snappydata/benchmark/snappy/tpch/QueryExecutor.scala [hunk 3]
```diff
@@ -125,28 +122,32 @@ object QueryExecutor {
...
     }
 
-    val queryStatisticsFileStream: FileOutputStream = new FileOutputStream(new File(queryStatisticsFileName))
+    val queryStatisticsFileStream: FileOutputStream =
+      new FileOutputStream(new File(queryStatisticsFileName))
     val queryStatisticsPrintStream: PrintStream = new PrintStream(queryStatisticsFileStream)
     queryStatisticsPrintStream.println(s"Iteration,ResponseTime")
     // scalastyle:off println
     try {
       println(s"Started executing $queryNumber")
```
Comment (nikhilbandi) [inline cluster/src/test/scala/io/snappydata/benchmark/snappy/tpch/QueryExecutor.scala]: Statement "Started executing query: $queryNumber" would be more clear.

File: cluster/src/test/scala/io/snappydata/benchmark/snappy/tpcds/TableCreationJob.scala [hunk 4]
```diff
@@ -31,81 +47,104 @@ object TableCreationJob extends SnappySQLJob{
...
+  def createColumnPartitionedTable(snappy: SnappySession,
+                                   props: Map[String, String], tableName: String): Unit = {
 
     val df = snappy.read.parquet(s"$dataLocation/$tableName")
+    snappy.dropTable(tableName, ifExists = true)
     snappy.createTable(tableName, "column",
       new StructType(df.schema.map(_.copy(nullable = false)).toArray), props)
     df.write.insertInto(tableName)
-    val cnt = df.collect().length
+    // val cnt = df.collect().length
```
Comment (nikhilbandi) [inline cluster/src/test/scala/io/snappydata/benchmark/snappy/tpcds/TableCreationJob.scala]: Please delete line if not required.

File: cluster/src/test/scala/io/snappydata/benchmark/snappy/tpch/QueryExecutor.scala (hunk 3 above)
Comment (cruizen) [reply cluster/src/test/scala/io/snappydata/benchmark/snappy/tpch/QueryExecutor.scala]: Okay.

File: cluster/src/test/scala/io/snappydata/benchmark/snappy/tpcds/TableCreationJob.scala (hunk 4 above)
Comment (cruizen) [reply cluster/src/test/scala/io/snappydata/benchmark/snappy/tpcds/TableCreationJob.scala]: The job ran very slow due to that old line of code. Kept it around to indicate that. Will add a comment.

**Pull request merged**